GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift

```diff
@@ -112,21 +116,27 @@ final class PeopleWithBonusesFetchController {
     }
 
     private func updatePersonWithImageFromNetwork(person: Person, completionBlock: (Result<Person>) -> Void) {
-        guard let requestTemplate = GravatarImageRequestTemplate(email: person.email) else {
-            completionBlock(.Failure(PeopleWithBonususFetchControllerError.IncorrectEmailAddress))
+
+        guard let avatarPath = person.avatarPath else {
+            completionBlock(.Failure(PeopleWithBonususFetchControllerError.NoAvatarURL))
+
             return
         }
 
-        requestSending.sendRequestForRequestTemplate(requestTemplate) { result in
+        dataDownloading.downloadDataAtPath(avatarPath) { result in
             switch result {
-            case .Success(let image):
-                completionBlock(.Success(person.copyWithImage(image)))
+            case .Success(let data):
+                if let image = UIImage(data: data) {
```

Comment:
What would you say at adding extension to DataDownloading instead?
```

public extension DataDownloading {
    func downloadImageAtPath(path: String, completion: (Result<UIImage>) -> Void) {
        downloadDataAtPath(path) { result in
            switch result {
            case .Success(let data):
                if let image = UIImage(data: data) {
                    completion(.Success(image))
                } else {
                    completion(.Failure(DataDownloadingError.ImageCannotBeFetched))
                }
            case .Failure(let error):
                completion(.Failure(error))
            }
        }
    }
}
```
and using here...